src/matrix_ordering.jl

```diff
@@ -0,0 +1,49 @@
+function pseudo_peripheral_node(g::SimpleGraph, src)
+    T = eltype(g)
+    u::T = src
+    lp = 0
+    while true
+        l = maximum(gdistances(g, u))
+        l <= lp && break
+        lp = l
+        for v in outneighbors(g, u)
+            if degree(g, v) < degree(g, u)
+               u = v
+            end
+        end
+    end
+    return u
+end
+
+function connected_rcm(g::SimpleGraph, visited, cm, src)
```

Comment:
This will modify `cm` and visited, so we should better name it `connected_rcm!`.